azurerm/resource_arm_sql_server_test.go

```diff
@@ -74,6 +75,23 @@ func TestAccAzureRMSqlServer_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMSqlServer_upperCaseName(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMSqlServer_upperCaseName(ri, testLocation())
```

Comment:
whilst this works, could we swap this out for some unit tests with both valid and invalid examples?, [ala this](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_servicebus_namespace_test.go#L67)?